internal/apidiff/apidiff.go

```diff
@@ -0,0 +1,259 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux darwin
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	osexec "os/exec"
+	"path"
+	"strings"
+)
+
+// TODO(noahdietz): remove this once the fix in golang.org/x/tools is released.
+// https://github.com/golang/go/issues/44796
+const ignored = "- MaxPublishRequestBytes: value changed from 0.000582077 to 10000000"
+const rootMod = "cloud.google.com/go"
+
+var repoMetadataPath string
+var verbose bool
+
+func init() {
+	flag.StringVar(&repoMetadataPath, "repo-metadata", "", "path to a repo-metadata-full JSON file [required]")
+	flag.BoolVar(&verbose, "verbose", false, "enable verbose command logging")
+}
+
+func main() {
+	flag.Parse()
+	if repoMetadataPath == "" {
+		log.Fatalln("Missing required flag: -repo-metadata")
+	}
+
+	head, err := exec("git", "log", "-1")
+	if err != nil {
+		log.Fatalln(err)
+	}
+	if strings.Contains(head, "BREAKING_CHANGE") {
+		log.Println("Not running apidiff because description contained tag BREAKING_CHANGE.")
+		return
+	}
+
+	root, err := os.Getwd()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	f, err := os.Open(repoMetadataPath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer f.Close()
+
+	var m manifest
+	if err := json.NewDecoder(f).Decode(&m); err != nil {
+		log.Fatalln(err)
+	}
+
+	_, err = exec("go", "install", "golang.org/x/exp/cmd/apidiff@latest")
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	diffs, diffingErrs, err := diffModules(root, m)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	if len(diffingErrs) > 0 {
+		fmt.Fprintln(os.Stderr, "The following packages encountered errors:")
+		for imp, err := range diffingErrs {
+			fmt.Fprintf(os.Stderr, "%s: %s\n", imp, err)
+		}
+	}
+
+	if len(diffs) > 0 {
+		fmt.Fprintln(os.Stderr, "The following breaking changes were found:")
+		for imp, d := range diffs {
+			fmt.Fprintf(os.Stderr, "%s:\n%s\n", imp, d)
+		}
+		os.Exit(1)
+	}
+}
+
+// manifestEntry is used for JSON marshaling in manifest.
+// Copied from internal/gapicgen/generator/gapics.go.
+type manifestEntry struct {
+	DistributionName  string `json:"distribution_name"`
+	Description       string `json:"description"`
+	Language          string `json:"language"`
+	ClientLibraryType string `json:"client_library_type"`
+	DocsURL           string `json:"docs_url"`
+	ReleaseLevel      string `json:"release_level"`
+}
+
+type manifest map[string]manifestEntry
+
+func diffModules(root string, m manifest) (map[string]string, map[string]error, error) {
+	diffs := map[string]string{}
+	issues := map[string]error{}
+
+	goCloud, err := cloneGoCloud()
```

Comment:
Could get rid of the `error` return if we clone GoCloud before calling this function. Doesn't really matter.

Reply:
Moved this into main and removed `cloneGoCloud`